### DISCUSSION TITLE
undefined symbol: 'csc_set_data'

### DISCUSSION THREAD

**Discussion by TRo202** [osqp/osqp#580] [C Interface]
Hi. 

I have a similar error with #545 and mine is `undefined symbol: 'csc_set_data'`.  I found that this function is defined in osqp_api_utils.h, and that header file is included in osqp.h if OSQP_EMBEDDED_MODE is not defined. 

So I was wondering if the reason why I am getting this error is that OSQP_EMBEDDED_MODE is pre-defined somewhere... But I don't know how to change the OSQP_EMBEDDED_MODE..

I tried it by just including it manually by adding `#include "osqp_api_utils.h"` line on my project header file, but I still get the same error.

Could you please help me?

**Comment by dg0yt** [comment]
Just reviewing some new vcpkg ports, and seeing that error for osqp-eigen on Android. It is tricky:
- `csc_set_data` is declared but not defined in this package.
  - It is used by osqp-eigen if not building that package with `OSQP_EIGEN_OSQP_IS_V1_FINAL`.
  - `OSQP_EIGEN_OSQP_IS_V1_FINAL` is enabled in osqp-eigen subject to try-compile tests.
  - These tests fail due to undefined symbols `osqp_start_interrupt_listener` etc.
- `osqp_start_interrupt_listener` is implemented here in `src/interrupt_unix.c` and `src/interrupt_windows.c`, but these sources are only compiled for `IS_MAC`, `IS_LINUX`, `IS_WINDOWS`.
- An immediate workaround is to build osqp with `OSQP_ENABLE_INTERRUPT` off.

**Reply by traversaro** [reply]
Good catch, that is definitely a osqp-eigen issue (osqp-eigen mantainer here). Feel free to open an issue in osqp-eigen repo about that, but as vcpkg only packages osqp v1.0.0, I think the fastest solution is just to explicitly set the `OSQP_IS_V1` and `OSQP_IS_V1_FINAL` CMake variables in osqp-eigen's vcpkg port, see https://github.com/gbionics/osqp-eigen/blob/19e1d8d1bfaa741922b0bbdd07352bf33a515ef5/cmake/OsqpEigenDependencies.cmake#L40-L48 .

**Reply by dg0yt** [reply]
Setting the V1 variables would just move the `osqp_start_interrupt_listener` issue from osqp-eigen configuration time to osgp-eigen build time.
The question is really if `src/interrupt_unix.c` can be applied to more than macOS and Linux.

**Reply by dg0yt** [reply]
And well, declaring but not defining `csc_set_data` is strange, too.
